fcgi.c

```diff
@@ -183,11 +185,43 @@ get_flags (FCGX_Request * r)
     return flags;
 }
 
+/* Implementation of
+   https://datatracker.ietf.org/doc/html/rfc3986#section-5.2.4
+   Most web servers will have already done this for us
+   but it is probably faster to do again then trying to
+   figure out if it is even needed.
+ */
+static char *
+normalise_path(const char *path)
+{
+    GString *normalised = g_string_new(NULL);
+    char *copy = g_strdup(path);
+    char *token = strtok((char *)copy, "/");
```

Comment:
strtok is regarded as dodgy in a threaded environment. you can use strtok_r instead.